unity-renderer/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/LoadableShape.cs

```diff
@@ -249,13 +260,24 @@ protected void OnLoadCompleted(LoadWrapper loadWrapper)
             OnFinishCallbacks = null;
         }
 
+        private void OnInsideBoundariesChanged(IDCLEntity entity, bool isInsideBoundaries)
+        {
+            UpdateAnimationStatus(entity, isInsideBoundaries);
+        }
+
         protected virtual void DetachShape(IDCLEntity entity)
         {
             if (entity == null || entity.meshRootGameObject == null)
                 return;
 
             LoadWrapper loadWrapper = Environment.i.world.state.GetLoaderForEntity(entity);
-            loadWrapper?.Unload();
+
+            if (loadWrapper != null)
+            {
+                entity.OnInsideBoundariesChanged -= s => OnInsideBoundariesChanged(entity, s);
```

Comment:
I don't think this unsubscription will actually work

Reply:
It doesnt, youc an use a local function